src/middlewares/filter.middleware.js

```diff
@@ -3,15 +3,14 @@ var filterService = require('../service/filterService')
 
 function addMetaFilters (req, res, next) {
   // If the request body has filter by metaFilter data, continue with the same filter, do not alter the values
-
-  // else call the getMetaSearchString() function to generate the search string for the meta filters
-  filterService.getMetaSearchString(function () {
+  // else call the getMetaSearchData() function to generate the JSON Object for the meta filters
+  filterService.getMetaSearchData(function () {
```

Comment:
does this need to be a callback based method?

Reply:
Not required, functionality has been implemented and tested locally, it is working fine.